packages/transmute-dashboard/src/components/Messages/Messages.js

```diff
@@ -0,0 +1,234 @@
+import React, { Component } from 'react';
+import AppBar from '../AppBar';
+
+import IPFS from 'ipfs';
+import Room from 'ipfs-pubsub-room';
+import _ from 'lodash';
+
+import { withStyles } from 'material-ui/styles';
+import Card, { CardActions, CardContent, CardHeader } from 'material-ui/Card';
+import Button from 'material-ui/Button';
+import Grid from 'material-ui/Grid';
+import { LinearProgress } from 'material-ui/Progress';
+import Input, { InputLabel } from 'material-ui/Input';
+import { FormControl } from 'material-ui/Form';
+import Select from 'material-ui/Select';
+import List, { ListItem, ListItemText } from 'material-ui/List';
+
+import { withAuth } from '@okta/okta-react';
+import config from '../../ipfs_pubsub_config';
+
+import theme from '../../theme';
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+  },
+  flex: {
+    flex: 1,
+  }
+});
+
+class Messages extends Component {
+  constructor(props) {
+    super(props);
+    this.ipfs = new IPFS(config);
+    this.state = {
+      info: null,
+      address: '',
+      message: '',
+      messages: [],
+      selectedPeer: {},
+      peers: {}
+    }
+
+    this.handleMessage = this.handleMessage.bind(this);
+    this.handleBroadcast = this.handleBroadcast.bind(this);
+    this.selectPeer = this.selectPeer.bind(this);
+    this.isMessageJSON = this.isMessageJSON.bind(this);
+    this.updatePeerStatus = this.updatePeerStatus.bind(this);
+  }
+
+  async componentWillMount() {
+    const user = await this.props.auth.getUser();
+    this.ipfs.once('ready', () => this.ipfs.id((err, info) => {
+      if (err) { throw err }
+      this.setState({ info });
+      
+      const map = {
+        id: info.id,
+        name: user.name,
+        pk: JSON.parse(user.did_document).id.split(':')[2],
```

Comment:
This is not always a public key, sometimes its a contract address. For true DID compatibility, we should be pulling from the publicKeys array, the key in hex, and the key type.

Reply:
I think its ok for now though

Reply:
I'll update with OpenPGP and using the pem.